### PR TITLE
fix(doc): add small size input example

### LINKED ISSUE #1229: Docs > Forms > Checks & radios Small Sizing example is not in v5 but was there in v4

### PR DESCRIPTION
**Pull request by hannahiss**
Closes #1229

### CODE REVIEW

**Review by louismaximepiton — COMMENTED** [review]
IMO, we should backport the associated CSS too, but we maybe need to warn about it with an ODS alert (as the small version isn't allowed by the brand) ?

**Comment by hannahiss** [comment]
I'm really not sure about the sizes of inputs, it is a proposition...

File: scss/forms/_form-control.scss [hunk 1]
```diff
@@ -131,7 +132,18 @@
...
 // Repeated in `_input_group.scss` to avoid Sass extend issues.
 
-// Boosted mod: no .form-control-sm
+.form-control-sm {
+  min-height: $input-height-sm;
+  padding: $input-padding-y-sm $input-padding-x-sm add($input-padding-y-sm, 1px); // Boosted mod
+  @include font-size($input-font-size-sm);
+  @include border-radius($input-border-radius-sm);
+
+  &::file-selector-button {
```
Comment (louismaximepiton) [inline scss/forms/_form-control.scss:141]: Is there any way to make it 30px high ?

File: scss/forms/_form-control.scss [hunk 2]
```diff
@@ -131,7 +132,18 @@
 //
 // Repeated in `_input_group.scss` to avoid Sass extend issues.
 
-// Boosted mod: no .form-control-sm
+.form-control-sm {
```
Comment (louismaximepiton) [inline scss/forms/_form-control.scss:135]: I think we should override the line-height value for sm input to make it 30px

File: scss/forms/_form-select.scss [hunk 3]
```diff
@@ -55,10 +55,16 @@
...
+  padding-top: $form-select-padding-y-sm;
+  padding-bottom: add($form-select-padding-y-sm, 1px); // Boosted mod
+  padding-left: $form-select-padding-x-sm;
+  @include font-size($form-select-font-size-sm);
+  @include border-radius($form-select-border-radius-sm);
+}
 
 .form-select-lg {
-  padding-top: subtract($form-select-padding-y-lg, 1px); // Boosted mod
+  padding-top: $form-select-padding-y-lg;
```
Comment (louismaximepiton) [inline scss/forms/_form-select.scss:67]: I think we should keep it as-is

File: scss/forms/_form-select.scss [hunk 4]
```diff
@@ -55,10 +55,16 @@
   }
 }
 
-// Boosted mod: no .form-select-sm
+.form-select-sm {
```
Comment (louismaximepiton) [inline scss/forms/_form-select.scss:58]: We maybe need to update the line-height to have 30px height

**Review by louismaximepiton — CHANGES_REQUESTED** [review]
I don't know if this is a good idea since we don't want to show people that they can be used, but I think we should display the small examples in the doc.

IMO, we should try to have every small input to have the same height : 30px or equivalent

**Review by louismaximepiton — APPROVED** [review]
LGTM, but we need to discuss a bit on:

* `$line-height-*` are weird to me (`$line-height-sm > $line-height-base`) but could be handled in another PR
* See on Bs why `.input-group` sizing changes padding of `.form-control` instead of having a `min-height:` like `.form-control-*` do.
* `.col-form-label-*` seems to be not aligned with the input value

**Comment by julien-deramond** [comment]
There are some conflicts to resolve here @hannahiss 🙏

**Comment by hannahiss** [comment]
Add small size examples to match Bootstrap examples, even if they are not part of ODS, in forms:
- Forms > Form control > Sizing
- Forms > Form control > File input
- Forms > Input group > Sizing
- Forms > Layout > Horizontal form label sizing
- Forms > Select > Sizing

Some callouts (warnings about incompatibility with ODS) should be added at all these places like in PR https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1614 (if PR https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1614 is merged before this one, should be added in this PR, if not, should be added in PR https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1614)

**Comment by hannahiss** [comment]
I updated the documentation with missing mentions to small sizes (compared to Bottstrap).

**Comment by julien-deramond** [comment]
Could you please update the migration guide as well?

**Comment by hannahiss** [comment]
> Could you please update the migration guide as well?

I'm wondering if you should mention it, because we don't want developers to use these small examples (not compatible with ODS) ! We may discuss it on daily ?

File: site/content/docs/5.2/migration.md [hunk 5]
```diff
@@ -75,6 +75,8 @@ If you need more details about the changes, please refer to the [v5.2.3 release]
 
 - <span class="badge bg-success">New</span> Form helpers icon buttons will help you to provide extra help information with the `.form-helper` utility class.
 
+- <span class="badge bg-success">New</span> Small size form controls are now available. Small inputs are defined by their `.form-control-sm`, small selects by their `.form-select-sm`. `.input-group-sm` can be used depending on the context while they can also be associated with `.col-form-label-sm` to correctly follow their size if needed.
```
Comment (julien-deramond) [inline site/content/docs/5.2/migration.md:78]: @hannahiss: I've added more info in this sentence to explain all new classes that have been added in this PR. Please check this out and suggest improvement if needed :)

**Review by julien-deramond — APPROVED** [review]
Pushed some minor modifications. LGTM!

File: site/content/docs/5.2/migration.md (hunk 5 above)
Comment (julien-deramond) [reply site/content/docs/5.2/migration.md]: Fixed via https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1230/commits/6d1f964d6f72971432a10c22dbc17e77a17cb131

**Pull request merged**